src/preact/base-element.js

```diff
@@ -784,6 +790,7 @@ export class PreactBaseElement extends AMP.BaseElement {
     const keys = Object.keys(current);
     for (let i = 0; i < keys.length; i++) {
       const key = keys[i];
+      // eslint-disable-next-line local/restrict-this-access
       wrapRefProperty(this, api, key);
```

Comment:
This only meta-programs over the `this.currentRef_` object, so it's fine.

Reply:
Is there a more general way to determine what kind of use is safe?